src/diffusers/pipelines/wuerstchen/pipeline_wuerstchen_prior.py

```diff
@@ -0,0 +1,307 @@
+                uncond_input.input_ids.to(device), attention_mask=uncond_input.attention_mask.to(device)
+            )
+
+            uncond_text_encoder_hidden_states = negative_prompt_embeds_text_encoder_output.last_hidden_state
+
+            # duplicate unconditional embeddings for each generation per prompt, using mps friendly method
+            seq_len = uncond_text_encoder_hidden_states.shape[1]
+            uncond_text_encoder_hidden_states = uncond_text_encoder_hidden_states.repeat(1, num_images_per_prompt, 1)
+            uncond_text_encoder_hidden_states = uncond_text_encoder_hidden_states.view(
+                batch_size * num_images_per_prompt, seq_len, -1
+            )
+            # done duplicates
+
+            # For classifier free guidance, we need to do two forward passes.
+            # Here we concatenate the unconditional and text embeddings into a single batch
+            # to avoid doing two forward passes
+            text_encoder_hidden_states = torch.cat([text_encoder_hidden_states, uncond_text_encoder_hidden_states])
+
+        return text_encoder_hidden_states
+
+    @torch.no_grad()
+    def __call__(
+        self,
+        prompt: Union[str, List[str]] = None,
+        height: int = 1024,
+        width: int = 1024,
+        num_inference_steps: Union[int, Dict[float, int]] = {2 / 3: 20, 0.0: 10},
+        guidance_scale: float = 8.0,
+        negative_prompt: Optional[Union[str, List[str]]] = None,
+        num_images_per_prompt: Optional[int] = 1,
+        generator: Optional[Union[torch.Generator, List[torch.Generator]]] = None,
+        latents: Optional[torch.FloatTensor] = None,
+        output_type: Optional[str] = "pt",
+        return_dict: bool = True,
+    ):
+        device = self._execution_device
+        do_classifier_free_guidance = guidance_scale > 1.0
+
+        if isinstance(num_inference_steps, int):
+            num_inference_steps = {0.0: num_inference_steps}
+
+        if isinstance(prompt, str):
+            prompt = [prompt]
+        elif not isinstance(prompt, list):
+            raise ValueError(f"`prompt` has to be of type `str` or `list` but is {type(prompt)}")
+
+        batch_size = len(prompt) if isinstance(prompt, list) else 1
+        text_encoder_hidden_states = self._encode_prompt(
+            prompt, device, num_images_per_prompt, do_classifier_free_guidance, negative_prompt
+        )
+
+        dtype = text_encoder_hidden_states.dtype
+        latent_height = ceil(height / 42.67)
+        latent_width = ceil(width / 42.67)
+        num_channels = self.prior.config.c_in
+        effnet_features_shape = (num_images_per_prompt * batch_size, num_channels, latent_height, latent_width)
+
+        self.scheduler.set_timesteps(num_inference_steps, device=device)
+        timesteps = self.scheduler.timesteps
+
+        latents = self.prepare_latents(effnet_features_shape, dtype, device, generator, latents)
+
+        for t in self.progress_bar(timesteps[:-1]):
```

Comment:
Our pipelines also include brief comments in the pipeline calls in a numbered manner so that that the code blocks are semantically well segregated. It also helps with readability. Could we also do that for these pipelines?